flycheck.el

```diff
@@ -3522,26 +3526,27 @@ Return the created overlay."
   ;; We must have a proper error region for the sake of fringe indication,
   ;; error display and error navigation, even if the highlighting is disabled.
   ;; We erase the highlighting later on in this case
-  (pcase-let* ((`(,beg . ,end) (flycheck-error-region-for-mode
-                                err (or flycheck-highlighting-mode 'lines)))
-               (overlay (make-overlay beg end))
-               (level (flycheck-error-level err))
-               (category (flycheck-error-level-overlay-category level)))
-    (unless (flycheck-error-level-p level)
-      (error "Undefined error level: %S" level))
-    (setf (overlay-get overlay 'flycheck-overlay) t)
-    (setf (overlay-get overlay 'flycheck-error) err)
-    ;; TODO: Consider hooks to re-check if overlay contents change
-    (setf (overlay-get overlay 'category) category)
-    (unless flycheck-highlighting-mode
-      ;; Erase the highlighting from the overlay if requested by the user
-      (setf (overlay-get overlay 'face) nil))
-    (when flycheck-indication-mode
-      (setf (overlay-get overlay 'before-string)
-            (flycheck-error-level-make-fringe-icon
-             level flycheck-indication-mode)))
-    (setf (overlay-get overlay 'help-echo) #'flycheck-help-echo)
-    overlay))
+  (when (eq (flycheck-error-buffer err) (current-buffer))
```

Comment:
@chrisdone That is not how the built-in error parsers behave currently; these always put the current buffer in `buffer` regardless of what file the error comes from.

So this pull request needs to include corresponding changes to built-in parsers, otherwise this change will break for all built-in checkers that report errors from multiple files, including Rust, C, C++, Fortran, and maybe others.

And it should include updates to the docstring of `flycheck-error` to clarify the relation between the filename and buffer slots now that the contents of the buffer slot have an actual effect on how Flycheck highlights errors.